### ISSUE TITLE
[bug] tensorflow inference container, python service is not enabled even though insert environment variable SAGEMAKER

### ISSUE THREAD

**Issue by sociengineer** [aws/deep-learning-containers#2692]
Checklist
- [x] I've prepended issue tag with type of change: [bug]
- [ ] (If applicable) I've attached the script to reproduce the bug
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [x] I'm using an existing DLC image listed here: https://docs.aws.amazon.com/deep-learning-containers/latest/devguide/deep-learning-containers-images.html
- [ ] I've built my own container based off DLC (and I've attached the code used to build my own image)

*Concise Description:*
In tensorflow inference contained, python service is not enabled even if I put environment variable as follows SAGEMAKER_SUBMIT_DIRECTORY = /opt/ml/model/code
SAGEMAKER_PROGRAM = inference.py

How can I override the INFERENCE PATH?

*DLC image/dockerfile: 763104351884.dkr.ecr.ap-northeast-2.amazonaws.com/tensorflow-inference:2.8-gpu

*Current behavior: python service is not enabled
https://github.com/aws/deep-learning-containers/blob/94df39f9bd4e9dc6a2a9c1a4f33d8cfd71fd1b7d/tensorflow/inference/docker/build_artifacts/sagemaker/serve.py#L136

*Expected behavior: python service is enabled 

*Additional context: 
I read the answer on AWS Answers about SAGEMAKER_SUBMIT_DIRECTORY Tensorflow inference container.              
https://answers.amazon.com/posts/336524
 
I want to make a finetuned tensorflow image classification model and deploy this model with customer inference code using CDK.
I followed the example code in SageMaker built-in tensorflow image classification code and wrought the equivalent CDK code using API.
https://github.com/aws/amazon-sagemaker-examples/blob/main/introduction_to_amazon_algorithms/image_classification_tensorflow/Amazon_TensorFlow_Image_Classification.ipynb
 
As I understood, it is impossible to deploy the fine-tuned model with custom inference code using the SAGEMAKER_SUBMIT_DIRECTORY pointing S3 URI. Because this algorithm is not XGB, SKL and Chainer.
When I checked the fine-tuned model that was created from SageMaker SDK, SAGEMAKER_SUBMIT_DIRECTORY was ‘/opt/ml/model/code’ and the extracted model.tar.gz folder had inference code was in the subdirectory of named code.
So, I repacked the fine-tuned model with the subdirectory including the custom inference code. However, the deployed model still ignores the custom inference code and response only probabilities without labels.
How to solve this problem? Is it only possible using SageMaker SDK, not API?
